### PR TITLE
enumerable methods

### PR DESCRIPTION
**Pull request by sjcco**
Micoverse 2nd ruby project

Recreated Enumerable methods

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello team** very nice job with this project, all your methods work as excepted, there is only one problem that needs to be fixed:
- [x] Your rubocop is failing locally currently you have 18 offenses, kindly fix those

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  end
+
+  # 6. my_none?
+  def my_none?(arg = nil)
+    if block_given?
+      !to_a.my_any?(&Proc.new)
+    else
+      !to_a.my_any?(arg)
+    end
+  end
```
Comment (enaburekhan) [inline enumerables.rb]: Four of the tests are failing due to this error: **tried to create Proc object without a block** from #my_none?.
And the error flags the **.new**
We can resolve this by rewriting the my_none? method. We can start like so:
def my_none?(arg = nil) my_each do |y|  if block_given? use the required if/elsif to check the given conditions... end end true end

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes requested:full_moon:
**Hi Carlos!
Great job on making the changes requested by the previous Tse:clap::clap:. However, we still need to make a few changes to get the project approved :muscle:**.

- [ ] enumerables #my_none? returns true if the block never returns true for all elements
- [ ] enumerables #my_none? returns false if the block ever returns true for all elements
- [ ] enumerables #my_none? returns false if the block ever returns true for all elements::range
- [ ] enumerables #my_none? does not mutate the original array
The above four tests are failing due to this error: **tried to create Proc object without a block**
As I already highlighted below in the pull request codebase(line 71 to 78). Rewrite the #my_none? using the suggested hints. 

Make the required changes and when you are done, request for another review.
Kindly drop your questions:point_down: if you have any and ping me on slack @ Eric Enaburekhan to respond.
Happy Coding:blush:
Eric Enaburekhan

**Review by GzimAsani — APPROVED** [review]
# **STATUS: APPROVED** 
Congratulations **TEAM**  , Great job implementing the requirements, this project is approved and you can move to your next project 
Don't forget to submit a project completion form.
 Cheers! 👍🏽

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Pull request merged**